modules/python/python-dest.c

```diff
@@ -177,14 +169,6 @@ _py_invoke_is_opened(PythonDestDriver *self)
   return _dd_py_invoke_bool_function(self, self->py.is_opened, NULL);
 }
 
-static void
-_py_invoke_retry_error(PythonDestDriver *self, PyObject *dict)
-{
-  if (!self->py.retry_error)
-    return;
-  _dd_py_invoke_void_function(self, self->py.retry_error, dict);
```

Comment:
As I see this is the only way for python binding to know syslog-ng gave up on a message. Maybe this can be used if user wants to implement a network like destination, and when they see the message is dropped due to server error (after 3 tries), they want to fallback to another server. Or just want to do their own logging on the failed message, like report an incident.

Reply:
Don't forget that we now have flush() as an interface (not yet exposed to Python, but I guess that's one of the next steps). It is in flush() where the plugin decides if a batch can be sent or not. It can even do its retry tracking logic (e.g. it can count the number of WORKER_INSERT_RESULT_ERROR return values in a row).

Logging can also happen in flush() or even during insert().

btw: I am not against the interface per-se, it was just difficult to retain in its current form because of batching. I've checked there were no important in-tree users, so I've dropped it. Reimplementing it would have increased the scope yet again and at 43 patches, this series is already huge.

Reply:
Ok sounds good to me. I do not have any other use case in mind.